storage/watchable_store.go

```diff
@@ -0,0 +1,335 @@
+package storage
+
+import (
+	"errors"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/storage/storagepb"
+)
+
+// EOF is the error returned by Watcher.Err when watcher reaches its end revision and
+// no more event is available.
+var EOF = errors.New("EOF")
```

Comment:
I tried and call it `ReachEnd`. limit could be confused with limit parameter in Range.